### DISCUSSION TITLE
Newbie issues

### DISCUSSION THREAD

**Discussion by glow12121** [labwc/labwc#978] [Q&A]
I have accumulated some questions about setting up LabWC... Would appreciate some help with these:

- [x] How do I get Alt+arrows to not move windows but have the default, say, Gnome behavior: navigate forward/back in Nautilus and other programs? SOLVED WITH `<keybind key="A-Left"><action name="None"/></keybind>` and `<keybind key="A-Right"><action name="None"/></keybind>` in `rc.xml`.
- [x] How to get Windows+up/down maximize/restore-to-previous-size a window? SOLVED WITH `<keybind key="W-Up"><action name="ToggleMaximize"/></keybind>`. 
- [ ] Is it possible to launch wofi with only the Super key? NOT YET.
- [x] A keybind to `wofi` command isn't enough to launch Wofi, unlike what is stated [here](https://labwc.github.io/getting-started.html). USE `wofi --show drun` INSTEAD.
- [ ] Is it possible to remove gaps and rounded corners for maximized windows? NOT YET
- [ ] Is it possible to completely remove windows' title-bars?
- [ ] How to refresh Waybar settings? (`Reconfigure` doesn't affect it.)
- [ ] How to make the Alt+Shift menu prettier? For example, display the programs' icons?
- [ ] How do I close windows with `W+q`?
- [x] Keybinds to switch to/move a window to workspace 1, 2 etc. ? BY ADDING TO `rc.xml`: `    <keybind key="W-1"><action name="GoToDesktop" to="1" wrap="yes" /></keybind>` etc. and `<keybind key="W-S-1"><action name="SendToDesktop" to="1" follow="no" wrap="yes" /></keybind>` etc.
- [x] `swaybg -i "/home/uuser/Pictures/flight_light.jpg" >/dev/null 2>&1 &` leads to a black screen instead of the picture. THERE WAS NO `swaybg` DEPENDENCY ALONGSIDE `labwc` IN FEDORA PACKAGE.
- [ ] How to scroll with the right edge of the trackpad?

**Comment by stefonarch** [comment]
This should be reverted to a [Discussion](https://github.com/labwc/labwc/discussions).
Did you read the documentation and rc.xml.all?  Most of your questions get answers there.

Just commenting out the A+arrow keybinds in `rc.xml,` this for the first question.

**Comment by glow12121** [comment]
OK, thanks, I'll go through `rc.xml.all` and address the Discussion if necessary.

**Comment by johanmalm** [comment]
Hey. Welcome :smile:

> How do I get Alt+arrows to not move windows but have the default, say, Gnome behavior: navigate forward/back in Nautilus and other programs?

You can use the `None` action, which might be more convenient than copying the whole `rc.xml.all`.

Something like this (from memory - not tested):

```
  <keyboard>
    <default/>
    <keybind key="A-Left"><action name="None"/></keybind>
  </keyboard>
```

> How to get Windows+up/down maximize/restore-to-previous-size a window?

`<keybind key="W-Up"><action name="ToggleMaximize"/></keybind>`

https://labwc.github.io/labwc-actions.5.html#entry_action_name=togglemaximize

> Is it possible to launch wofi with only the Super key?

No. I'm afraid we've not implemented that yet. On the TODO list.

> A keybind to `wofi` command isn't enough to launch Wofi, unlike what is stated [here](https://labwc.github.io/getting-started.html).

Yes, I think it's `wofi --show run`

> Is it possible to remove gaps and rounded corners for maximized windows?

Almost. `<action name="SnapToEdge" direction="center" />` respects gaps.

> Is it possible to completely remove windows' title-bars?

Yes. Action `ToggleDecoration` (but it removes the border too atm).

#876 is nearly there to keep borders.

> How to refresh Waybar settings? (`Reconfigure` doesn't affect it.)

Not sure :smile: Different project. Doesn't it respect `killall -s SIGHUP waybar`?

> How to make the Alt+Shift menu prettier? For example, display the programs' icons?

You can edit https://github.com/labwc/labwc/blob/e06a8cfc7acabfd6e52916371d22a30bd0c7b72a/docs/menu.xml#L5 to change the entries. Someone submitted a pretty menu with font-icons using fontawesome (or similar), but I can't find it now.

For anything else, it's a matter for rolling ones sleeves up and submit patches I'm afraid :smile:

> How do I close windows with `W+q`?

Use the `Close` action.

> Keybinds to switch to/move a window to workspace 1, 2 etc. ?

https://github.com/labwc/labwc/blob/e06a8cfc7acabfd6e52916371d22a30bd0c7b72a/docs/rc.xml.all#L84-L86

> `swaybg -i "/home/uuser/Pictures/flight_light.jpg" >/dev/null 2>&1 &` leads to a black screen instead of the picture.

Two `u` in `user`??

> How to scroll with the right edge of the trackpad?

IIRC libinput does not support it.

**Reply by glow12121** [reply]
Many thanks! I've made a task list in the opening post... will update it more this week.

**Reply by glow12121** [reply]
> > Is it possible to remove gaps and rounded corners for maximized windows?
> 
> Almost. `<action name="SnapToEdge" direction="center" />` respects gaps.

How do I include it in `rc.xml` as the default look?

> > Is it possible to completely remove windows' title-bars?
> 
> Yes. Action `ToggleDecoration` (but it removes the border too atm).
> 
> #876 is nearly there to keep borders.

Again, how do I include it in `rc.xml` as the default look?

\* \* \*

When opening `wofi --show drun` with W+d several times, the launcher opens several instances on top of one another. Pressing Escape seemingly does nothing. Pressing Enter with a blinking cursor closes the instances one after another.

Just sharing an idea. It would make sense to add `wlsunset` to `autostart`. For example: `wlsunset -l 55.7 -L 37.6 -t 3700 -T 6000 >/dev/null 2>&1 &`

Swaybg doesn't work with `fill` option: `swaybg -i "/home/uuser/MEGA/Pictures/Обои/flight_light.jpg" fill >/dev/null 2>&1 &`

**Comment by johanmalm** [comment]
Step 3 of https://labwc.github.io/getting-started.html is updated with a `wofi --show drun` example :smile:

**Comment by Consolatis** [comment]
> > How to scroll with the right edge of the trackpad?
> 
> IIRC libinput does not support it.

https://wayland.freedesktop.org/libinput/doc/latest/configuration.html#scrolling
> The scroll method can be chosen or disabled altogether but most devices only support a subset of available scroll methods. libinput’s default is two-finger scrolling for multi-touch touchpads, edge scrolling for single-touch touchpads. On tracksticks, button scrolling is enabled by default.

See also https://wayland.freedesktop.org/libinput/doc/latest/scrolling.html#scrolling
I am not sure if labwc supports configuring the mode though.

> Is it possible to remove gaps and rounded corners for maximized windows?

Removing rounded corners on maximize is on the todo list for a while already but its not implemented so far.
Regarding the gaps: there should be none when the window is actually maximized (vs "fake maximized" like with `SnapToEdge center` or a region that is at `0, 0, 100%, 100%`).

**Reply by glow12121** [reply]
In regards to trackpad edge scrolling, here is respective config section from Sway. Not sure if it is relevant:

```
input 2:7:SynPS/2_Synaptics_TouchPad {
	}

input * {
	tap enabled
	# scroll_method on_button_down
	# scroll_button BTN_BACK
	scroll_method edge
	}
```

**Comment by stefonarch** [comment]
>How do I close windows with W+q?
```
    <keybind key="W-q">
      <action name="Close"/>
    </keybind>
```
Using Super key for any combination is in contradiction for using it alone for opening wofi or other stuff, unless there is  a workaround in place, like "action on release only".

**Comment by stefonarch** [comment]
> > > Is it possible to completely remove windows' title-bars?
> > 
> > 
> > Yes. Action `ToggleDecoration` (but it removes the border too atm).
> > #876 is nearly there to keep borders.
> 
> Again, how do I include it in `rc.xml` as the default look?
> 
Try with
```
    <windowRule title="*" >
      <action name="ToggleDecoration"/>
    </windowRule>
```

> Swaybg doesn't work with `fill` option: `swaybg -i "/home/uuser/MEGA/Pictures/Обои/flight_light.jpg" fill >/dev/null 2>&1 &`


```
 -m, --mode             Set the mode to use for the image.
  -o, --output           Set the output to operate on or * for all.
  -v, --version          Show the version number and quit.

Background Modes:
  stretch, fit, fill, center, tile, or solid_color
````

` -m fill`

**Reply by glow12121** [reply]
Why `<actn name="ToggleAlwaysOnBottom"/>`? (spelling of `actn` and the option itself?

**Reply by stefonarch** [reply]
sorry, copy and paste error and typos... should be ok now. 
Quoting selected text is broken in discussions, it copies always the whole posting.

**Reply by glow12121** [reply]
Thanks!

**Comment by johanmalm** [comment]
>  Is it possible to remove gaps and rounded corners for maximized windows?

In a `maximized` state gaps are not shown.
